Backport selinux change from zed

priteau opened the pull request:
The disable-selinux role has been renamed to selinux and now supports setting desired state.

Previously Kayobe was defaulting to disabling and rebooted the host - to avoid audit logs filling up. This change allows operators to define desired SELinux state and defaults to permissive - to adhere to those site policies that require SELinux to be at least in permissive state.

Note: unlike the original patch, this backport keeps the default selinux state as disabled.

Change-Id: I42933b0b7d55c69c9f6992e331fafb2e6c42d4d1 (cherry picked from commit caa7cc54ee5919b1b74dcdff21850051641eb9b5)

markgoddard commented on ansible/roles/selinux/defaults/main.yml:
```diff
@@ -1,7 +1,13 @@
+
 # Whether to reboot to apply SELinux config changes.
-disable_selinux_do_reboot: true
+selinux_do_reboot: false
```
How about this to retain backwards compat?
```suggestion
# Whether to reboot to apply SELinux config changes.
disable_selinux_do_reboot: true
selinux_do_reboot: "{{ disable_selinux_do_reboot }}"
```

markgoddard commented on ansible/selinux.yml:
```diff
@@ -0,0 +1,9 @@
+    - selinux
+  roles:
+    - role: selinux
+      selinux_reboot_timeout: "{{ 600 if ansible_facts.virtualization_role == 'host' else 300 }}"
```
```suggestion
      selinux_reboot_timeout: "{{ disable_selinux_reboot_timeout | default(600 if ansible_facts.virtualization_role == 'host' else 300) }}"
```

priteau committed:
Backport selinux change from zed

The disable-selinux role has been renamed to selinux and now supports
setting desired state.

Previously Kayobe was defaulting to disabling and rebooted the host - to
avoid audit logs filling up. This change allows operators to define
desired SELinux state and defaults to permissive - to adhere to those
site policies that require SELinux to be at least in permissive state.

Note that unlike the original patch, this backport keeps the default
selinux state as disabled.

Co-authored-by: Mark Goddard <mark@stackhpc.com>
Change-Id: I42933b0b7d55c69c9f6992e331fafb2e6c42d4d1
(cherry picked from commit caa7cc54ee5919b1b74dcdff21850051641eb9b5)

MoteHue commented:
Merging now so I can use at AZ
